test/cpp/interop/xds_interop_server.cc

```diff
@@ -65,13 +66,59 @@ using grpc::testing::SimpleResponse;
 using grpc::testing::TestService;
 using grpc::testing::XdsUpdateHealthService;
 
+namespace {
+constexpr absl::string_view kRpcBehaviorMetadataKey = "rpc-behavior";
+constexpr absl::string_view kErrorCodeRpcBehavior = "error-code-";
+
+std::set<std::string> GetRpcBehaviorMetadata(ServerContext* context) {
+  std::set<std::string> rpc_behaviors;
```

Comment:
I don't think using a set here is correct. According to the gRFC: "The `rpc-behavior` header value can have multiple options separated by commas. In that case, the value should be split by commas and the *options should be applied in the order specified*". Specifically, I think using a set here loses the ordering of the list separated by commas.

Reply:
https://github.com/grpc/grpc-java/blob/42b4c61d5e869b1534808204bc71df84720f5087/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java#L403 see here for deterministic list to build out.

Reply:
Sure. Please review: https://github.com/grpc/grpc/pull/34164

Reply:
Thanks! Reviewed and approved. This keeps the ordering, I commented same thing on in flight Go PR: https://github.com/grpc/grpc-go/pull/6575#discussion_r1304954335 :).